### DISCUSSION TITLE
What is `Adapter concurrency`?

### DISCUSSION THREAD

**Discussion by aldur11** [Koenkk/zigbee2mqtt#10664] [Devices]
Does anyone know what `Adapter concurrency` does? All I can find in the docs is a brief description about setting it and not what it really does.

**Comment by Koenkk** [comment]
It control the amount of commands that the adapter can handle at the same time. e.g. if it is set to 2 and you send a turn on command to 3 bulbs; the first 2 will be send immediately, the last one will be send whenever the first or second command has completed.

**Reply by aldur11** [reply]
Very interesting, would this setting is most useful for large deployments?

Also, the setting says the default of null uses the recommended values, what are the recommended values? (I'm using CC26X2R1 for my 3 coordinators)

**Reply by Koenkk** [reply]
CC26X2R1 has 16 by default (see https://www.zigbee2mqtt.io/guide/configuration/adapter-settings.html#transmitter-power `adapter_concurrent`

**Reply by aldur11** [reply]
Can it be set higher, or is the default the max? I have some light fixtures with 15 bulbs and I've noticed that most of the bulbs will turn on at the same time, then there is a few that lag behind for about 8 seconds.

**Reply by Koenkk** [reply]
You can set it as high as you want, not sure if it will improve your situation, probably better to put them in groups: https://www.zigbee2mqtt.io/guide/usage/groups.html

**Reply by aldur11** [reply]
Oh wow, that is exactly what I needed! Thanks so much for the pointer.

**Comment by dhawalx** [comment]
@Koenkk Can you explain what it does by means of zigbee messages transmission to nodes ?? 
or 
it also related to receiving multiple zigbee messages from router at same time? 

Does it has any impact on if 40 devices in network sending zigbee messages to coordinator at same time?

**Reply by Koenkk** [reply]
This is only about the sending part, receiving messages is not impacted by this setting
